.github/workflows/eden_test.yml

```diff
@@ -0,0 +1,20 @@
+---
+name: Eden Test from lf-edge/eden
+
+on:
```

Comment:
Would it be possible with these reusable workflows to specify that smoke tests (job: smoke) should be run already on opening PR, while other tests require approval?
I think it would be great improvement over the current setup, because now even for basic eden tests we have to wait for somebody to approve PR. @uncleDecart WDYT?

Reply:
@milan-zededa 

**TL;DR** In theory we can. But it could be tricky and will look like magic outside Hogwarts  

So we will need to actually create reusable workflow running smoke tests and another running all other tests and condition it on running after successful completion of smoke tests. This raises couple of questions:
- Will it be run after approval for first time committers? Most likely, yes. Otherwise people can ddos you with random PRs triggering tests which will use up all your resources
- Will there be problems with LTS releases? Possibly. Right now we are having troubles with `assets.yml` being run from master branch on LTS. It uses same mechanism I described for smoke tests. This could be a problem if we change tests. We can solve latter with reusable workflows for `assets.yml`, but conditioning on jobs are a lil bit tricky. We can use something like,

```
      - name: Network test
        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
        uses: ....
```
but it might give us overall green light when tests are skipped

Reply:
Regarding DoS - I don't think there would be any difference between smoke tests job and say PR build job.
And first time committers only get DCO check without approval I believe, so this should no be a problem.